### ISSUE TITLE
Don't validate SAM transformed resources for rule I3042

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#2483]
### CloudFormation Lint Version

v0.71.1

### What operating system are you using?

Mac

### Describe the bug

When SAM transforms templates it can create hardcoded ARNs based on its scenario. It would make sense to not validate those ARNs against rule I3042

### Expected behavior

To not raise I3042 on resources that are created by SAM transform.

### Reproduction template

```yaml

```
